modules/pixi/PixiLayerRapidOverlay.js

```diff
@@ -0,0 +1,166 @@
+ * @class
+ */
+export class PixiLayerRapidOverlay extends AbstractLayer {
+
+  /**
+   * @constructor
+   * @param  scene    The Scene that owns this Layer
+   * @param  layerID  Unique string to use for the name of this Layer
+   */
+  constructor(scene, layerID) {
+    super(scene, layerID);
+    this._clear();
+    this._enabled = true;
+
+    const overlays = new PIXI.Container();
+    overlays.name = `${this.layerID}-overlays`;
+    overlays.sortableChildren = false;
+    overlays.interactiveChildren = true;
+    this.overlaysContainer = overlays;
+
+
+    const basemapContainer = this.scene.groups.get('basemap');
+    basemapContainer.addChild(overlays);
+  }
+
+
+  /**
+   * render
+   * Render the GeoJSON custom data
+   * @param  frame      Integer frame being rendered
+   * @param  viewport   Pixi viewport to use for rendering
+   * @param  zoom       Effective zoom to use for rendering
+   */
+  render(frame, viewport, zoom) {
+    if (!this.enabled || !(this.hasData())) return;
+
+    const vtService = this.context.services.vectortile;
+    const datasets = this.context.systems.rapid.datasets;
+
+    // for (const [key, dataset] in datasets) {
+
+    // }
+
+    const customColor = new PIXI.Color(datasets.get('fbRoads').color);
+    const overlay = datasets.get('fbRoads').overlay;
+    let overlayData = [];
+    if (overlay && vtService) {   // fetch data from vector tile service
+      if ((zoom >= overlay.minZoom ) && (zoom <= overlay.maxZoom)) {  // avoid firing off too many API requests 
+        vtService.loadTiles(overlay.url);
+      }
+      overlayData = vtService.getData(overlay.url).map(d => d.geojson);
+    }
+
+    // const polygons = overlayData.filter(d => d.geometry.type === 'Polygon' || d.geometry.type === 'MultiPolygon');
+    // const lines = overlayData.filter(d => d.geometry.type === 'LineString' || d.geometry.type === 'MultiLineString');
+    const points = overlayData.filter(d => d.geometry.type === 'Point' || d.geometry.type === 'MultiPoint');
+
+    this.renderPoints(frame, viewport, zoom, points, customColor);
+  }
+
+
+  /**
+   * renderPoints
+   * @param  frame      Integer frame being rendered
+   * @param  viewport   Pixi viewport to use for rendering
+   * @param  zoom       Effective zoom to use for rendering
+   * @param  lines      Array of point data
+   * @param  color      The color to use
+   */
+  renderPoints(frame, viewport, zoom, points, color) {
+    const parentContainer = this.overlaysContainer;
+
+    //Extremely inefficient but we're not drawing anything else at this zoom
+    parentContainer.removeChildren();
+
+    for (const d of points) {
+      const parts = (d.geometry.type === 'Point') ? [d.geometry.coordinates]
+        : (d.geometry.type === 'MultiPoint') ? d.geometry.coordinates : [];
+
+      for (let i = 0; i < parts.length; ++i) {
+        const coords = parts[i];
+
+          const xyCoords = viewport.project(coords);
+          const feature = new PIXI.Graphics().
```

Comment:
This is wasteful- we should save these circles as a texture and then use that texture, instead of repeating the same draw call over and over.